tests/expectations/metrics/column_map_metrics/test_unexpected_indices_and_query.py

```diff
@@ -0,0 +1,298 @@
+
+    unexpected_index_list: MetricConfiguration = MetricConfiguration(
+        metric_name="column_values.in_set.unexpected_index_list",
+        metric_domain_kwargs={"column": "animals"},
+        metric_value_kwargs=metric_value_kwargs,
+    )
+    unexpected_index_list.metric_dependencies = {
+        "unexpected_condition": unexpected_columns_metric,
+        "table.columns": table_columns_metric,
+    }
+    results: Dict[Tuple[str, str, str], MetricValue] = engine.resolve_metrics(
+        metrics_to_resolve=(unexpected_index_list,), metrics=metrics
+    )
+    for (key, val) in results.items():
+        assert val == [{"pk_1": 3}, {"pk_1": 4}, {"pk_1": 5}]
+
+
+@pytest.mark.unit
+def test_sa_unexpected_index_list_metric_with_id_pk(
+    sa, animal_table_df, metric_value_kwargs_complete
+):
+    df: pd.DataFrame = animal_table_df
+    metric_value_kwargs: dict = metric_value_kwargs_complete
+
+    engine: SqlAlchemyExecutionEngine = build_sa_engine(df=df, sa=sa)
+    (
+        table_columns_metric,
+        unexpected_columns_metric,
+        metrics,
+    ) = _build_table_columns_and_unexpected(engine, metric_value_kwargs_complete)
+
+    unexpected_index_list = MetricConfiguration(
+        metric_name="column_values.in_set.unexpected_index_list",
+        metric_domain_kwargs={"column": "animals"},
+        metric_value_kwargs=metric_value_kwargs,
+    )
+    unexpected_index_list.metric_dependencies = {
+        "unexpected_condition": unexpected_columns_metric,
+        "table.columns": table_columns_metric,
+    }
+    results: Dict[Tuple[str, str, str], MetricValue] = engine.resolve_metrics(
+        metrics_to_resolve=(unexpected_index_list,), metrics=metrics
+    )
+    for (key, val) in results.items():
+        assert val == [{"pk_1": 3}, {"pk_1": 4}, {"pk_1": 5}]
+
+
+@pytest.mark.unit
+def test_sa_unexpected_index_list_metric_without_id_pk(sa, animal_table_df):
+    df: pd.DataFrame = animal_table_df
+    metric_value_kwargs: dict = {
+        "value_set": ["cat", "fish", "dog"],
+        "parse_strings_as_datetimes": False,
+        "result_format": {
+            "result_format": "COMPLETE",
+            "partial_unexpected_count": 20,
+            "include_unexpected_rows": False,
+        },
+    }
+
+    engine: SqlAlchemyExecutionEngine = build_sa_engine(df=df, sa=sa)
+    (
+        table_columns_metric,
+        unexpected_columns_metric,
+        metrics,
+    ) = _build_table_columns_and_unexpected(engine, metric_value_kwargs)
+
+    unexpected_index_list = MetricConfiguration(
+        metric_name="column_values.in_set.unexpected_index_list",
+        metric_domain_kwargs={"column": "animals"},
+        metric_value_kwargs=metric_value_kwargs,
+    )
+    unexpected_index_list.metric_dependencies = {
+        "unexpected_condition": unexpected_columns_metric,
+        "table.columns": table_columns_metric,
+    }
+    with pytest.raises(MetricResolutionError):
```

Comment:
Ah took a minute but I understand now - this MetricResolutionError is raised if we are trying to retrieve the unexpected_index_list metric without specifying `unexpected_index_column_names`.